[Bug]: TimeSeries.get_timestamps fails for numpy array timestamps

### What happened?

I was trying to use the new convenience method on the TimeSeries class, `get_timestamps`, and it threw an error on my time series, which use numpy arrays for timestamps. 

### Steps to Reproduce

```python
time_series = mock_TimeSeries(data=[1, 2, 3], timestamps=np.array([3, 4, 5]), rate=None)
assert_array_equal(time_series.get_timestamps(), [3, 4, 5])
```

### Traceback

```python
def get_timestamps(self):
        """
        Get the timestamps of this TimeSeries. If timestamps are not stored in this TimeSeries, generate timestamps.
        """
>       if self.fields.get('timestamps'):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E       ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

src/pynwb/base.py:396: ValueError
```

### Operating System

macOS

### Python Executable

Conda

### Python Version

3.13

### Package Versions

_No response_

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [x] Have you checked the [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured this bug was not already [reported](https://github.com/NeurodataWithoutBorders/pynwb/issues)?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,7 +13,8 @@
 - Added `get_starting_time()` and `get_duration()` methods to `Units` to get the earliest spike time and total duration (span from earliest to latest spike) across all units. @h-mayorquin [#2164](https://github.com/NeurodataWithoutBorders/pynwb/pull/2164)
 
 ### Fixed
-- Fixed invalid CSS properties in documentation assistant toggle that prevented proper positioning on displays ≥1400px wide. @rly [#2151](https://github.com/NeurodataWithoutBorders/pynwb/pull/2151) 
+- Fixed invalid CSS properties in documentation assistant toggle that prevented proper positioning on displays ≥1400px wide. @rly [#2151](https://github.com/NeurodataWithoutBorders/pynwb/pull/2151)
+- Fixed `TimeSeries.get_timestamps()` to handle numpy array timestamps when they are set. @pauladkisson [#2181](https://github.com/NeurodataWithoutBorders/pynwb/pull/2181)
 
 ### Changed
 - Added Python 3.14 support. @bendichter, @rly [#2168](https://github.com/NeurodataWithoutBorders/pynwb/pull/2168)

--- a/src/pynwb/base.py
+++ b/src/pynwb/base.py
@@ -393,7 +393,7 @@ class TimeSeries(NWBDataInterface):
         """
         Get the timestamps of this TimeSeries. If timestamps are not stored in this TimeSeries, generate timestamps.
         """
-        if self.fields.get('timestamps'):
+        if self.fields.get('timestamps') is not None:
             return self.timestamps
         else:
             return np.arange(len(self.data)) / self.rate + self.starting_time

--- a/src/pynwb/base.py
+++ b/src/pynwb/base.py
@@ -447,7 +447,7 @@ class TimeSeries(NWBDataInterface):
         if self.num_samples == 1:
             return 0.0
 
-        if self.fields.get('timestamps'):
+        if self.fields.get('timestamps') is not None:
             timestamps = self.timestamps
             return float(timestamps[-1] - timestamps[0])
         else:

--- a/tests/unit/test_base.py
+++ b/tests/unit/test_base.py
@@ -429,6 +429,9 @@ class TestTimeSeries(TestCase):
         time_series = mock_TimeSeries(data=[1, 2, 3], timestamps=[3, 4, 5], rate=None)
         assert_array_equal(time_series.get_timestamps(), [3, 4, 5])
 
+        time_series = mock_TimeSeries(data=[1, 2, 3], timestamps=np.array([3, 4, 5]), rate=None)
+        assert_array_equal(time_series.get_timestamps(), [3, 4, 5])
+
     def test_get_data_in_units(self):
         ts = mock_TimeSeries(data=[1., 2., 3.], conversion=2., offset=3.)
         assert_array_equal(ts.get_data_in_units(), [5., 7., 9.])

--- a/tests/unit/test_base.py
+++ b/tests/unit/test_base.py
@@ -580,6 +580,9 @@ class TestTimeSeries(TestCase):
         # Duration from 0 to 3 seconds = 3 seconds
         self.assertEqual(ts.get_duration(), 3.0)
 
+        ts = mock_TimeSeries(data=[1, 2, 3, 4], timestamps=np.array([0.0, 1.0, 2.0, 3.0]))
+        self.assertEqual(ts.get_duration(), 3.0)
+
     def test_get_duration_single_sample(self):
         """Test get_duration with single sample returns 0"""
         ts_rate = mock_TimeSeries(data=[1], rate=10.0)

